include/FilesystemManagement.h

```diff
@@ -0,0 +1,144 @@
+    /// @remarks Any value larger than success should be considered an error/failure. Any function that returns
+    /// such a value did not complete and in most cases did not modify the filesystem.
+    enum class ModifyResult
+    {
+        Success = 0,       ///< Everything worked.
+        // Existence Errors
+        AlreadyExists,     ///< The file or directory to be created already exists.
+        DoesNotExist,      ///< The file of directory specified wasn't found.
+        // Path Errors
+        InvalidPath,       ///< The path provided is invalid. Could be caused by a "." at the end.
+        LoopingPath,       ///< The path provided contains links that form a loop.
+        NameTooLong,       ///< The name of a file or directory specified is too long.
+        // Permission Errors
+        PermissionDenied,  ///< The filesystem permissions forbid the action to be taken.
+        ReadOnly,          ///< The filesystem only permits read operations.
+        // Directory Errors
+        NotADirectory,     ///< The path was expected to direct to a directory, but doesn't.
+        IsADirectory,      ///< The path was expected to direct to non-directory link, but does.
+        NotEmpty,          ///< The directory specified is not empty.
+        // Other Errors
+        IOError,           ///< There was a hardware I/O error while completing the operation.
+        NoSpace,           ///< There is insufficient space on the filesystem to complete the operation.
+        MaxLinksExceeded,  ///< The operation would exceeded the maximum links supported for the directory.
+        CurrentlyBusy,     ///< The file or directory specified is in use by the system and operation can't complete.
+        OperationCanceled, ///< The operation was canceled or aborted.
+        // Unknown
+        Unknown            ///< Error is unknown.
+    };
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Basic File Management
+
+    /// @brief Verifies the existence of a file.
+    /// @param FilePath The path and name of the file to check for.
+    /// @return Returns true if the file at the specified location exists, false otherwise.
+    Boole MEZZ_LIB FileExists(const StringView FilePath);
+
+    /// @brief Copies a file on disk to a new location.
+    /// @note This function makes no attempt to copy file permissions or attributes, only data.
+    /// @param OldFilePath The existing path to the file (including the filename) to be copied.
+    /// @param NewFilePath The path (including the filename) to where the file should be copied.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully copied, false otherwise.
+    ModifyResult MEZZ_LIB CopyFile(const StringView OldFilePath, const StringView NewFilePath, const Boole FailIfExists);
```

Comment:
This is easy for a gamedev to ignore. If it works most of the time we should presume any will time pressure will ignore it and this will lead to silent errors. These errors are needless, either these should be [[nodiscard]] or throw exceptions on failures. Since this is already written I think these should just be tagged [[nodiscard]].

Reply:
Added [[nodiscard]] to all the functions.